doc/releases/changelog-dev.md

```diff
@@ -4,192 +4,331 @@
+  print(qml.draw(clifford_circuit)())
   ```
-  
+
   ```pycon
-  >>>draw
-  "0: ──RX──H──┤  <Z>\n1: ──H───RY─┤     \n2: ──RZ─────┤     "
+  0: ──RZ(-2.36)──H────╭●──S†─────────H──RZ(0.10)──H─╭●──S──────────H──S──T──GlobalPhase(-2.36)─┤
+  1: ──RZ(-0.79)──H──S─╰X──RZ(-0.10)─────────────────╰X──RZ(-1.57)──H──T─────GlobalPhase(-2.36)─┤
   ```
 
-* Measurement statistics can now be collected for mid-circuit measurements. Currently,
-  `qml.expval`, `qml.var`, `qml.probs`, `qml.sample`, and `qml.counts` are supported on
-  `default.qubit`, `default.mixed`, and the new `DefaultQubit2` device.
+  `RZ` gates are approximated in the Clifford+T basis using the method 
+  described in [Ross and Selinger (2016)](https://arxiv.org/abs/1403.2975). The 
+  `epsilon` argument in `clifford_t_decomposition` allows for the error in approximating
+  `RZ` gates to be adjusted.
+
+<h4>Postselection and statistics in mid-circuit measurements 📌</h4>
+
+* TODO postselection
+
+* Measurement statistics can now be collected for mid-circuit measurements.
   [(#4544)](https://github.com/PennyLaneAI/pennylane/pull/4544)
 
   ```python
-  dev = qml.device("default.qubit", wires=2)
+  dev = qml.device("default.qubit")
 
   @qml.qnode(dev)
   def circ(x, y):
       qml.RX(x, wires=0)
       qml.RY(y, wires=1)
       m0 = qml.measure(1)
-      return qml.expval(qml.PauliZ(0)), qml.sample(m0)
+      return qml.expval(qml.PauliZ(0)), qml.expval(m0), qml.sample(m0)
   ```
 
-  QNodes can be executed as usual when collecting mid-circuit measurement statistics:
-
   ```pycon
-  >>> circ(1.0, 2.0, shots=5)
-  (array(0.6), array([1, 1, 1, 0, 1]))
+  >>> circ(1.0, 2.0, shots=10000)
+  (0.5606, 0.7089, array([0, 1, 1, ..., 1, 1, 1]))
   ```
+  
+  Support is provided for both
+  [finite-shot and analytic modes](https://docs.pennylane.ai/en/stable/introduction/circuits.html#shots)
+  and devices default to using the
+  [deferred measurement](https://docs.pennylane.ai/en/stable/code/api/pennylane.defer_measurements.html)
+  principle to enact the mid-circuit measurements.
 
-* Operator transforms `qml.matrix`, `qml.eigvals`, `qml.generator`, and `qml.transforms.to_zx` are updated
-  to the new transform program system.
-  [(#4573)](https://github.com/PennyLaneAI/pennylane/pull/4573)
+  In future releases, we will be exploring the ability to combine and manipulate mid-circuit
+  measurements such as `qml.expval(m0 @ m1)` or `qml.expval(m0 @ qml.PauliZ(0))`.
 
-* Transforms can be applied on devices following the new device API.
- [(#4667)](https://github.com/PennyLaneAI/pennylane/pull/4667)
+<h4>Exponentiate Hamiltonians with flexible Trotter products 🐖</h4>
 
-* All gradient transforms are updated to the new transform program system.
- [(#4595)](https://github.com/PennyLaneAI/pennylane/pull/4595)
+* Higher-order Trotter-Suzuki methods are now easily accessible through a new operation
+  called `TrotterProduct`.
+  [(#ABCD)]()
```

Comment:
Just putting a note here that the PR numbers need to be added once they're merged